commit f60c4bb9d944849d3d9269a3f4c0782353c973ec
Author: Prashant Nevase <pnevase@microsoft.com>
Date:   Fri May 8 05:37:26 2026 -0700

    Prefix dark mode decision-tree threshold constants with kFeature.
    
    This CL prefixes feature constants with kFeature to clearly
    distinguish them from other constants, improving readability and
    scalability as more features are getting added.
    
    Bug: 449909524
    Change-Id: I84492594bead3e3b26a926933edfa11cbdba4c4b
    Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/7831584
    Commit-Queue: Stephen Chenney <schenney@chromium.org>
    Reviewed-by: Stephen Chenney <schenney@chromium.org>
    Auto-Submit: Prashant Nevase <pnevase@microsoft.com>
    Cr-Commit-Position: refs/heads/main@{#1627580}

diff --git a/third_party/blink/renderer/platform/graphics/dark_mode_image_classifier.cc b/third_party/blink/renderer/platform/graphics/dark_mode_image_classifier.cc
index 8151543ee33f..1d1b7fdf2348 100644
--- a/third_party/blink/renderer/platform/graphics/dark_mode_image_classifier.cc
+++ b/third_party/blink/renderer/platform/graphics/dark_mode_image_classifier.cc
@@ -17,23 +17,25 @@
 namespace blink {
 namespace {
 
-// Decision tree lower and upper thresholds for grayscale and color images.
-constexpr std::array<float, 2> kLowColorCountThreshold = {0.8125, 0.015137};
-constexpr std::array<float, 2> kHighColorCountThreshold = {1, 0.025635};
-
+const int kMaxSampledPixels = 1000;
+const int kMaxBlocks = 10;
+const float kMinOpaquePixelPercentageForForeground = 0.2;
 // Color is considered light if its luma is above this threshold.
 constexpr int kHighLightnessThreshold = 96;
 
-// Luminance ratio threshold above which the image is classified as light.
-constexpr float kHighLuminanceThreshold = 0.5f;
+// Decision tree thresholds for classifying images
+
+// Lower and upper color thresholds for grayscale and color images.
+constexpr std::array<float, 2> kFeatureLowColorCountThreshold = {0.8125,
+                                                                 0.015137};
+constexpr std::array<float, 2> kFeatureHighColorCountThreshold = {1, 0.025635};
 
 // Transparency ratio threshold above which the image is classified as
 // transparent, meaning most of the background pixels are transparent.
-constexpr float kTransparencyRatioThreshold = 0.4f;
+constexpr float kFeatureTransparencyRatioThreshold = 0.4f;
 
-const int kMaxSampledPixels = 1000;
-const int kMaxBlocks = 10;
-const float kMinOpaquePixelPercentageForForeground = 0.2;
+// Luminance ratio threshold above which the image is classified as light.
+constexpr float kFeatureHighLuminanceThreshold = 0.5f;
 
 bool IsColorGray(const SkColor& color) {
   return abs(static_cast<int>(SkColorGetR(color)) -
@@ -288,20 +290,20 @@ DarkModeResult DarkModeImageClassifier::ClassifyUsingDecisionTree(
   // foreground is predominantly light. Inverting such images would make
   // visible pixels darker, causing them to merge with dark background, so do
   // not apply filter.
-  if (features.transparency_ratio > kTransparencyRatioThreshold &&
-      features.high_luminance_ratio > kHighLuminanceThreshold) {
+  if (features.transparency_ratio > kFeatureTransparencyRatioThreshold &&
+      features.high_luminance_ratio > kFeatureHighLuminanceThreshold) {
     return DarkModeResult::kDoNotApplyFilter;
   }
 
   // Very few colors means it's not a photo, apply the filter.
   if (features.color_buckets_ratio <
-      kLowColorCountThreshold[features.is_colorful]) {
+      kFeatureLowColorCountThreshold[features.is_colorful]) {
     return DarkModeResult::kApplyFilter;
   }
 
   // Too many colors means it's probably photorealistic, do not apply it.
   if (features.color_buckets_ratio >
-      kHighColorCountThreshold[features.is_colorful]) {
+      kFeatureHighColorCountThreshold[features.is_colorful]) {
     return DarkModeResult::kDoNotApplyFilter;
   }
 
